demos/link-prediction/random-walks/ctdne-link-prediction.ipynb

```diff
@@ -0,0 +1,680 @@
+      "text/plain": [
+       "<Figure size 504x504 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "temporal_node_embeddings = temporal_model.wv.vectors\n",
+    "static_node_embeddings = static_model.wv.vectors\n",
+    "plot_tsne(\"TSNE visualisation of temporal node embeddings\", temporal_node_embeddings)\n",
+    "plot_tsne(\"TSNE visualisation of static node embeddings\", static_node_embeddings)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Link Prediction using Node Embeddings\n",
+    "\n",
+    "The node embeddings we've generated can now be used as input for a link prediction task. The reported AUC scores from the reference paper for this dataset are:\n",
+    "\n",
+    "* Node2Vec: 0.759 \n",
+    "* CTDNE: 0.777\n",
+    "\n",
+    "Below are a set of helper functions we can use to train and evaluate a link prediction classifier. The rest of the notebook will use the binary operator defined in the cell below. Available operators in the demo are:\n",
```

Comment:
I'm not personally a huge fan of adding unused code to notebooks: it's easy for the code to just be a distraction from the teaching, and it's not tested and so may break. We could delete it, but that's a bit less nice than exploring multiple styles, especially if they give very different results.

Do you think we could at least run the temporal embeddings ROC AUC evaluation with all of them, but maybe still use only L2 for static and the t-SNE plot?

Reply:
Some of the operators ended up causing convergence warnings while training the classifier (although the scores still looked fine) so that's partly why I didn't go through all of them - should I just remove them?

Reply:
Looks like it would be `l1` and `l2` - I'll switch to just using both of those unconditionally

Reply:
The unconditional switch ended up looking a lot more verbose and I wasn't such a fan of introducing even more helper functions in the notebook.. so I ended up just going with `l2`. The scores end up looking pretty much identical for `l1` and `l2` too